pkg/sql/sem/builtins/aggregate_builtins.go

```diff
@@ -1580,6 +1609,174 @@ func (a *arrayAggregate) Size() int64 {
 	return sizeOfArrayAggregate
 }
 
+type aggStatementStatistics struct {
+	singleDatumAggregateBase
+
+	stats              appstatspb.StatementStatistics
+	firstJsonDatumSize int64
+}
+
+func newAggStatementStatistics(
+	params []*types.T, evalCtx *eval.Context, _ tree.Datums,
+) eval.AggregateFunc {
+	return &aggStatementStatistics{
+		singleDatumAggregateBase: makeSingleDatumAggregateBase(evalCtx),
+	}
+}
+
+// Add the statistics information into a single object.
+func (a *aggStatementStatistics) Add(ctx context.Context, datum tree.Datum, _ ...tree.Datum) error {
+	if datum != nil && a.firstJsonDatumSize == 0 {
+		a.firstJsonDatumSize = int64(datum.Size())
+		err := a.updateMemoryUsage(ctx, a.firstJsonDatumSize)
+		if err != nil {
+			return err
+		}
+	}
+
+	return mergeStatementStatsHelper(&a.stats, datum)
```

Comment:
More for my own understanding again, why do we not update memory usage while in the helper functions? Is the idea that we're only creating a single object per helper function call which subsequently gets garbage collected?